homework/index.js

```diff
@@ -29,25 +32,150 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  // 1- create function to add row
+  function createRow(table, label, content) {
+    const row = createAndAppend('tr', table);
+    createAndAppend('th', row, { text: label });
+    createAndAppend('td', row, { text: content });
+    return row;
+  }
+
+  // 2- create details of repositories
+  function renderRepoDetails(detailsSection, repo) {
+    const li = createAndAppend('table', detailsSection, { class: 'item' });
+    const table = createAndAppend('table', li);
+    // 2-1 add repository name
+    const repoName = createRow(table, 'Repository:', '');
+    createAndAppend('a', repoName.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+    // 2-2 add description
+    if (repo.description) {
+      createRow(table, 'Description:', repo.description);
+    } else {
+      createRow(table, 'Description:', 'N/A');
+    }
+    // 2-3 add Forks
+    createRow(table, 'Forks:', repo.forks);
+    // 2-4 add updated
+    createRow(table, 'Updated:', new Date(repo.updated_at).toLocaleString());
+  }
+
+  // 3- Create function to sort the repositories
+  function sorting(part1, part2) {
+    return part1.name.localeCompare(part2.name);
+  }
+
+  // 4- Creating the contributions part
+  function createContributors(contributorSection, url) {
+    createAndAppend('h4', contributorSection, {
+      text: 'Contributions: ',
+    });
+    fetchJSON(url)
+      .then(contributors => {
+        if (contributors && contributors.length) {
+          const ul = createAndAppend('ul', contributorSection, {
+            class: 'contributor-list',
+          });
+          contributors.forEach(contributor => {
+            const contributorItem = createAndAppend('li', ul, {
+              class: 'contributor-item',
+            });
+            const contributorLink = createAndAppend('a', contributorItem, {
+              href: contributor.html_url,
+              target: '_blank',
+            });
+            const contributorDiv = createAndAppend('div', contributorLink, {
+              class: 'contributor',
+            });
+            createAndAppend('img', contributorDiv, {
+              src: contributor.avatar_url,
+              class: 'contributor-avatar',
+            });
+            const contributorDetails = createAndAppend('div', contributorDiv, {
+              class: 'contributor-details',
+            });
+            createAndAppend('div', contributorDetails, {
+              text: contributor.login,
+              class: 'contributor-login',
+            });
+            createAndAppend('div', contributorDetails, {
+              text: contributor.contributions,
+              class: 'contributions-number',
+            });
+          });
```

Comment:
This function is getting fairly large. It would be better to put the code from the arrow of the `.forEach()` (lines 83-107) into a separate, named function and use it instead of the arrow function:

```js
function renderContributor(contributor, ul) {
  ...
}

function createContributors(contributorSection, url) {
  ...
  contributors.forEach(contributor => renderContributor(contributor, ul));
  ...
}
```